### PR TITLE
Laboratory work #3, Loreleya Gellertova - 23FPL1

### PR DESCRIPTION
**Pull request by LoraGellertova**

### CODE REVIEW

File: lab_3_ann_retriever/main.py [hunk 1]
```diff
@@ -104,6 +105,17 @@ def tokenize(self, text: str) -> list[str] | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(text, str):
+            return None
+        punctuation = ("!'\"#$%&()*+,-./:;<=>?@[]^_`{|}~1234567890"
```
Comment (EIzharikov) [inline lab_3_ann_retriever/main.py:110]: is numbers a pucntuation too?

File: lab_3_ann_retriever/main.py [hunk 2]
```diff
@@ -117,6 +129,10 @@ def tokenize_documents(self, documents: list[str]) -> list[list[str]] | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        token_dict = []
+        for text in documents:
```
Comment (EIzharikov) [inline lab_3_ann_retriever/main.py:133]: return as list comprehension

File: lab_3_ann_retriever/main.py [hunk 3]
```diff
@@ -130,6 +146,22 @@ def _remove_stop_words(self, tokens: list[str]) -> list[str] | None:
...
+        if len(tokens) == 0 or len(self._stop_words) == 0:
+            return None
+        for elem in self._stop_words:
+            if not isinstance(elem, str):
+                return None
+        for elem in tokens:
+            if not isinstance(elem, str):
+                return None
+
+        for elem in self._stop_words:
```
Comment (EIzharikov) [inline lab_3_ann_retriever/main.py:160]: go list comprehension

File: lab_3_ann_retriever/settings.json [hunk 4]
```diff
@@ -1,3 +1,3 @@
 {
-    "target_score": 0
+    "target_score": 4
```
Comment (EIzharikov) [inline lab_3_ann_retriever/settings.json:2]: good start, but you should hurry up! Make CI green, fix all comments and continue your work. You can reach 8!

File: lab_3_ann_retriever/main.py (hunk 1 above)
Comment (LoraGellertova) [reply lab_3_ann_retriever/main.py:110]: no, but the task asks us to remove digits too

File: lab_3_ann_retriever/main.py (hunk 2 above)
Comment (LoraGellertova) [reply lab_3_ann_retriever/main.py:133]: done

File: lab_3_ann_retriever/main.py (hunk 3 above)
Comment (LoraGellertova) [reply lab_3_ann_retriever/main.py:160]: done

File: lab_3_ann_retriever/settings.json (hunk 4 above)
Comment (LoraGellertova) [reply lab_3_ann_retriever/settings.json:2]: in progress (thank you)

File: lab_3_ann_retriever/start.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 # pylint:disable=duplicate-code, too-many-locals, too-many-statements, unused-variable
 from pathlib import Path
+from lab_3_ann_retriever.main import Tokenizer
```
Comment (EIzharikov) [inline lab_3_ann_retriever/start.py:7]: show example at least for 4

File: lab_3_ann_retriever/main.py [hunk 6]
```diff
@@ -130,6 +149,18 @@ def _remove_stop_words(self, tokens: list[str]) -> list[str] | None:
...
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(tokens, list) or not isinstance(self._stop_words, list):
+            return None
+        if len(tokens) == 0 or len(self._stop_words) == 0:
+            return None
+        for elem in self._stop_words:
+            if not isinstance(elem, str):
+                return None
+        for elem in tokens:
```
Comment (EIzharikov) [inline lab_3_ann_retriever/main.py:159]: is it necessarily? leave only the necessary checks

**Review by EIzharikov — COMMENTED** [review]
go 6!!! You can do it. Fix CI and continue your work!

**Comment by EIzharikov** [comment]
Nothing new to review. Good luck for Friday

**Pull request closed without merging**